buildSrc/src/main/kotlin/otel.java-conventions.gradle.kts

```diff
@@ -175,7 +167,7 @@ dependencies {
   dependencyManagement(platform(project(":dependencyManagement")))
   afterEvaluate {
     configurations.configureEach {
-      if (isCanBeResolved && !isCanBeConsumed) {
+      if (!isCanBeConsumed && this != dependencyManagement) {
```

Comment:
This change is required for the composite build to work:

So this is not enough to make composite builds work as there is at least one other problem, namely how dependency management is set up. The constraints seem to be only applied if the ultimately built project has the constraints applied. For example, without this change, if I try inside the examples subdirectory:

```
$ ./gradlew classes --include-build ..

...

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':opentelemetry-examples-zipkin:compileJava'.
> Could not resolve all task dependencies for configuration ':opentelemetry-examples-zipkin:compileClasspath'.
   > Could not find io.zipkin.reporter2:zipkin-reporter:.
     Required by:
         project :opentelemetry-examples-zipkin > project :odin-java:exporters:opentelemetry-exporter-zipkin
```

We can see that `opentelemetry-exporter-zipkin` has the dependency `api("io.zipkin.reporter2:zipkin-reporter")`. However, the api configuration extends from nothing, so it does not have dependency management applied, and opentelemetry-examples-zipkin:compileClasspath does not use the java-conventions plugin so it does not have dependency management applied either.